textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -116,18 +116,22 @@ private void writeTableInfoToCatalog(String tableName, String indexDirectory, Sc
         // write table catalog
         DataStore tableCatalogStore = new DataStore(CatalogConstants.TABLE_CATALOG_DIRECTORY,
                 CatalogConstants.TABLE_CATALOG_SCHEMA);
+        DataWriter dataWriter = new DataWriter(tableCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer());
         insertTupleToDirectory(tableCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(),
-                CatalogConstants.getTableCatalogTuple(tableName, indexDirectory, luceneAnalyzerString));
+                CatalogConstants.getTableCatalogTuple(tableName, indexDirectory, luceneAnalyzerString), dataWriter);
+        dataWriter.close();
+        dataWriter = null;
        
         // write schema catalog
         Schema tableSchema = Utils.getSchemaWithID(schema);
-
         DataStore schemaCatalogStore = new DataStore(CatalogConstants.SCHEMA_CATALOG_DIRECTORY,
                 CatalogConstants.SCHEMA_CATALOG_SCHEMA);
+        dataWriter = new DataWriter(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer());
         // each attribute in the table schema will be one row in schema catalog
         for (ITuple tuple : CatalogConstants.getSchemaCatalogTuples(tableName, tableSchema)) {
-            insertTupleToDirectory(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(), tuple);
+            insertTupleToDirectory(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(), tuple,dataWriter);
```

Comment:
@zuozhiw Please review these changes that moved "dataWriter" into RelationManger.